src/server/frontend_wayland/window_wl_surface_role.cpp

```diff
@@ -383,12 +394,47 @@ mir::shell::SurfaceSpecification& mf::WindowWlSurfaceRole::spec()
     return *pending_changes;
 }
 
+void mf::WindowWlSurfaceRole::apply_pending_size_bounds()
+{
+    if (weak_scene_surface.lock())
```

Comment:
Is there any particular reason to do this check? You obviously don't *use* the `scene::Surface` here, so this must be here purely for optimisation (ie: we don't need to do the following if the surface has died)?

It's not entirely obvious to me that this will *be* an optimisation, as it introduces a bunch of atomic operations.

Reply:
It's not an optimization. It's so we use `params` if the surface hasn't been created yet. It is common throughout this file to take alternative paths before and after the surface has been created.